GVFS/GVFS.Mount/InProcessMount.cs

```diff
@@ -550,7 +552,21 @@ private void MountAndStartWorkingDirectoryCallbacks(CacheServerInfo cache)
                 this.tracer.RelatedInfo("Git status cache is not enabled");
             }
 
-            this.fileSystemCallbacks = this.CreateOrReportAndExit(() => new FileSystemCallbacks(this.context, this.gitObjects, RepoMetadata.Instance, virtualizer, gitStatusCache), "Failed to create src folder callback listener");
+            this.gvfsDatabase = this.CreateOrReportAndExit(() => new GVFSDatabase(this.context.FileSystem, this.context.Enlistment.EnlistmentRoot, new SqliteDatabase()), "Failed to create database connection");

```

Comment:
In my testing on the perf machine which has 36 cores it was only using 4 connections so I'm not sure we should open and keep around a bunch of connections that we will never use.  A new connection will get created and added back into the pool if there aren't any available to use so I'm not sure creating a bunch at the beginning has much benefit.  If we only set this to 1 but the projection change needs more it will expand the pool anyways so I found 5 to be a good number.  Let me know if you feel differently.

Reply:
> In my testing on the perf machine which has 36 cores it was only using 4 connections so I'm not sure we should open and keep around a bunch of connections that we will never use.

In those tests how many threads were creating files in parallel?

> A new connection will get created and added back into the pool if there aren't any available to use so I'm not sure creating a bunch at the beginning has much benefit.

That's a good point, it would be a one time hit as new connections are added and it looks like from that point on the connections stay around?  If that's the case and you've seen good perf with 5 then I agree there's no need to change this.

Reply:
>In those tests how many threads were creating files in parallel?

Creating temp files runs 
`int numThreads = Math.Max(8, Environment.ProcessorCount);`

And the recursive read does
```
string filereadArgs = string.Format(
	"{0} --threads={1}",
	Path.Combine(this.WorkingDirectory, path),
	Environment.ProcessorCount * 1.5);
```